File: src/main/java/com/tradey/technical_analysis/domain/entity/OHLCV.java
package com.tradey.technical_analysis.domain.entity;

import lombok.Builder;
import lombok.Getter;

@Builder
@Getter
public class OHLCV {
    private String symbol;
    private String timestamp;
    private double open;
    private double high;
    private double low;
    private double close;
    private double volume;

    private Double ma50;
    private Double ma200;
    private Double diffMa50Ma200;
}

File: src/main/java/com/tradey/technical_analysis/infrastructure/repositories/dynamo/dto/ohlcv.py
from decimal import Decimal
from typing import Optional
from pydantic import BaseModel

from src.domain.entity.ohlcv import OHLCV


class DynamoOHLCVDTO(BaseModel):
    symbol: str
    open: Decimal
    high: Decimal
    low: Decimal
    close: Decimal
    volume: Decimal
    timestamp: str

    ma50: Optional[Decimal] = None
    ma200: Optional[Decimal] = None
    diff_ma50_ma200: Optional[Decimal] = None

    def to_model(self) -> OHLCV:
        return OHLCV.model_validate(self.model_dump())
